relay-quotas/src/redis.rs

```diff
@@ -200,19 +511,36 @@ impl RedisRateLimiter {
                 let retry_after = self.retry_after(REJECT_ALL_SECS);
                 rate_limits.add(RateLimit::from_quota(quota, &item_scoping, retry_after));
             } else if let Some(quota) = RedisQuota::new(quota, item_scoping, timestamp) {
-                // Remaining quotas are expected to be trackable in Redis.
-                let key = quota.key();
-                let refund_key = get_refunded_quota_key(&key);
-
-                invocation.key(key);
-                invocation.key(refund_key);
-
-                invocation.arg(quota.limit());
-                invocation.arg(quota.expiry().as_secs() + GRACE);
-                invocation.arg(quantity);
-                invocation.arg(over_accept_once);
-
-                tracked_quotas.push(quota);
+                if quota.scope == QuotaScope::Global {
```

Comment:
the condition here is a bit ugly, as it's just incidental that the new global scope always uses the budget cache. We talked briefly before about if we could use the budget system for all ratelimits, but i have yet to explore that possibility.